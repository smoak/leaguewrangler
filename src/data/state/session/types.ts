export interface Session {
  readonly userToken: string;
}
