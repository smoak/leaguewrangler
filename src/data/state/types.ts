import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
