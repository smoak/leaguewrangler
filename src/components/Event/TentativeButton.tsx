import React, { FC } from 'react';
import { IconButton } from '@material-ui/core';
import TentativeIcon from '@material-ui/icons/HelpOutlineOutlined';
import RsvpedTentativeIcon from '@material-ui/icons/Help';

interface TentativeButtonProps {
  readonly isTentative: boolean;
  readonly className: string;
}

const TentativeButton: FC<TentativeButtonProps> = ({ className, isTentative }) => {
  const icon = isTentative ? <RsvpedTentativeIcon className={className} /> : <TentativeIcon className={className} />;
  return <IconButton aria-label="Tentative">{icon}</IconButton>;
};

export default TentativeButton;
