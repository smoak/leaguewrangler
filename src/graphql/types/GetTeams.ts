/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetTeams
// ====================================================

export interface GetTeams_teams {
  __typename: "Team";
  name: string;
  id: number;
}

export interface GetTeams {
  /**
   * Get teams for a user
   */
  teams: (GetTeams_teams | null)[];
}
