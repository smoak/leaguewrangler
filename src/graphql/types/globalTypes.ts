export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
};

export type Query = {
  __typename?: 'Query';
  /** Gets the current user */
  currentUser: User;
  /** Get a single event */
  event?: Maybe<Event>;
  /** Get events for a team */
  events: Array<Maybe<Event>>;
  /** Get teams for a user */
  teams: Array<Maybe<Team>>;
};


export type QueryEventArgs = {
  eventId: Scalars['Int'];
  teamId: Scalars['Int'];
};


export type QueryEventsArgs = {
  teamId: Scalars['Int'];
};


export type QueryTeamsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** A user */
export type User = {
  __typename?: 'User';
  displayName: Scalars['String'];
  emailAddress1: Scalars['String'];
  events: Array<Maybe<Event>>;
  firstName: Scalars['String'];
  fullName: Scalars['String'];
  lastName: Scalars['String'];
  profilePhotoFullUrl: Scalars['String'];
  profilePhotoSmallUrl: Scalars['String'];
  profilePhotoThumbnailUrl: Scalars['String'];
  userId: Scalars['Int'];
};


/** A user */
export type UserEventsArgs = {
  dashboardOnly?: InputMaybe<Scalars['Boolean']>;
};

/** An event */
export type Event = {
  __typename?: 'Event';
  endTimestamp?: Maybe<Scalars['Int']>;
  eventId: Scalars['Int'];
  eventType: Scalars['String'];
  location?: Maybe<Location>;
  seasonId: Scalars['Int'];
  seasonName?: Maybe<Scalars['String']>;
  startTimestamp?: Maybe<Scalars['Int']>;
  status: Scalars['String'];
  team: Team;
  title: Scalars['String'];
  viewerRsvpStatus: RsvpStatus;
};

/** A location */
export type Location = {
  __typename?: 'Location';
  address?: Maybe<Scalars['String']>;
  googleMapsDirectionsUrl?: Maybe<Scalars['String']>;
  googleMapsUrl?: Maybe<Scalars['String']>;
  isPublic: Scalars['Boolean'];
  locationId: Scalars['Int'];
  name: Scalars['String'];
};

/** A team */
export type Team = {
  __typename?: 'Team';
  name: Scalars['String'];
  photoFullUrl?: Maybe<Scalars['String']>;
  photoSmallUrl?: Maybe<Scalars['String']>;
  photoThumbnailUrl?: Maybe<Scalars['String']>;
  shortName: Scalars['String'];
  teamId: Scalars['Int'];
};

/** The RSVP status of a user */
export enum RsvpStatus {
  /** Indicates the user is available to attend */
  AVAILABLE = 'AVAILABLE',
  /** Indicates the user might attend */
  MAYBE = 'MAYBE',
  /** Indicates the user will not attend */
  NO = 'NO',
  /** Indicates the user has not RSVPed */
  NONE = 'NONE',
  /** Indicates the user will attend */
  YES = 'YES'
}

export type Mutation = {
  __typename?: 'Mutation';
  /** Create a user token */
  createUserToken?: Maybe<UserToken>;
  /** RSVP to an event */
  saveRsvp: Event;
};


export type MutationCreateUserTokenArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


export type MutationSaveRsvpArgs = {
  comments?: InputMaybe<Scalars['String']>;
  eventId: Scalars['Int'];
  rsvpStatus: RsvpStatus;
  teamId: Scalars['Int'];
};

/** A user token */
export type UserToken = {
  __typename?: 'UserToken';
  /** A unique token for the user matched. Tokens are 36-character, lower-cased GUIDs. */
  token: Scalars['String'];
};
