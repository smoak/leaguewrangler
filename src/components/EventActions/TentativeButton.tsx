import { IconButton } from '@mui/material';
import TentativeIcon from '@mui/icons-material/HelpOutlineOutlined';
import RsvpedTentativeIcon from '@mui/icons-material/HelpTwoTone';
import { FC, useCallback } from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

interface TentativeButtonProps {
  readonly isTentative: boolean;
  readonly className: string;
  onRsvp(rsvpStatus: RsvpStatus): void;
}

const TentativeButton: FC<TentativeButtonProps> = ({ onRsvp, className, isTentative }) => {
  const rsvpCallback = useCallback(() => {
    onRsvp(RsvpStatus.MAYBE);
  }, [onRsvp]);

  const icon = isTentative ? (
    <RsvpedTentativeIcon className={className} color="primary" />
  ) : (
    <TentativeIcon className={className} />
  );
  return (
    <IconButton aria-label="Tentative" onClick={rsvpCallback} size="large">
      {icon}
    </IconButton>
  );
};

export default TentativeButton;
