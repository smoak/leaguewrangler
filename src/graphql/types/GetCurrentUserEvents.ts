/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RsvpStatus } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetCurrentUserEvents
// ====================================================

export interface GetCurrentUserEvents_currentUser_events_team {
  __typename: "Team";
  shortName: string;
  photoThumbnailUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events_location {
  __typename: "Location";
  name: string;
  address: string | null;
  googleMapsUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events {
  __typename: "Event";
  eventId: number;
  title: string;
  endTimestamp: number | null;
  startTimestamp: number | null;
  viewerRsvpStatus: RsvpStatus;
  team: GetCurrentUserEvents_currentUser_events_team;
  location: GetCurrentUserEvents_currentUser_events_location | null;
}

export interface GetCurrentUserEvents_currentUser {
  __typename: "User";
  events: (GetCurrentUserEvents_currentUser_events | null)[];
}

export interface GetCurrentUserEvents {
  /**
   * Gets the current user
   */
  currentUser: GetCurrentUserEvents_currentUser;
}
