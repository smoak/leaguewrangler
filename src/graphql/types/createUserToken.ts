import * as Types from './globalTypes';

export type CreateUserTokenMutationVariables = Types.Exact<{
  username: Types.Scalars['String'];
  password: Types.Scalars['String'];
}>;


export type CreateUserTokenMutation = { __typename?: 'Mutation', createUserToken?: { __typename?: 'UserToken', token: string } | null };
