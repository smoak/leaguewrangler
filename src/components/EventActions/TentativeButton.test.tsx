import { IconButton } from '@material-ui/core';
import { ShallowWrapper, shallow } from 'enzyme';
import React from 'react';

import { RsvpStatus } from '../../graphql/types/globalTypes';

import TentativeButton from './TentativeButton';

describe('TentativeButton', () => {
  let component: ShallowWrapper;

  describe('when tentative', () => {
    beforeEach(() => {
      component = shallow(<TentativeButton isTentative={true} className="test" onRsvp={jest.fn()} />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when not tentative', () => {
    beforeEach(() => {
      component = shallow(<TentativeButton isTentative={false} className="test" onRsvp={jest.fn()} />);
    });

    it('renders as expected', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the user rsvps', () => {
    let onRsvp: jest.Mock;

    beforeEach(() => {
      onRsvp = jest.fn();
      component = shallow(<TentativeButton isTentative={true} className="test" onRsvp={onRsvp} />);

      component.find(IconButton).simulate('click');
    });

    it('calls `onRsvp` with MAYBE', () => {
      expect(onRsvp).toHaveBeenCalledWith(RsvpStatus.MAYBE);
    });
  });
});
