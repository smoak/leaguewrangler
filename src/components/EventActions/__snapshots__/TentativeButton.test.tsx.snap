// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TentativeButton when not tentative renders as expected 1`] = `
<WithStyles(IconButton)
  aria-label="Tentative"
  onClick={[Function]}
>
  <pure(HelpOutlineOutlinedIcon)
    className="test"
  />
</WithStyles(IconButton)>
`;

exports[`TentativeButton when tentative renders as expected 1`] = `
<WithStyles(IconButton)
  aria-label="Tentative"
  onClick={[Function]}
>
  <pure(HelpTwoToneIcon)
    className="test"
    color="primary"
  />
</WithStyles(IconButton)>
`;
